src/connector/src/sink/encoder/text.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::catalog::Schema;
+use risingwave_common::types::ToText;
+
+use super::RowEncoder;
+
+pub struct TextEncoder {
+    pub schema: Schema,
+    // the column must contain only one element
+    pub col_indices: Option<Vec<usize>>,
+}
+
+impl TextEncoder {
+    pub fn new(schema: Schema, col_indices: Option<Vec<usize>>) -> Self {
+        Self {
```

Comment:
Prefer to restrict the data type here:
* allowed: bool, int16, int32, int64, varchar, serial, int256
* denied
  * float32, float64
    * `1e6::float4` can be either `1e+06` or `1000000`
  * interval, struct, list, bytea
    * PostgreSQL `ToText` unfriendly to downstream
  * jsonb
    * #14442
* maybe
  * decimal
    * `1.0` and `1.00` are equal as `decimal` but unequal as text
  * date
    * `-0001-01-01` is same as `0002-01-01 BC`
  * time, timestamptz
    * `12:34:56.020` is same as `12:34:56.02` (i.e. pad to 3/6/9 or not)
  * timestamp
    * `2024-01-01 12:34:56` is neither ISO 8601 nor RFC 3339 but a mixture of both.